scripts/uberenv/packages/mfem/package.py

```diff
@@ -497,10 +499,16 @@ def find_optional_library(name, prefix):
                 ld_flags_from_dirs([spec['gslib'].prefix.lib], ['gs'])]
 
         if '+netcdf' in spec:
+            netcdf_libdirs = [spec['netcdf-c'].prefix.lib]
```

Comment:
These were required if linking a static netcdf into mfem, it required the transitive hdf5 brought along for the ride.

Reply:
I'd like Greg to look at this section before i throw it up on a Spack PR.  Since there are like 3 ways to build this list in the mfem package.

Reply:
@white238 I think that mostly looks fine, I'd prefer if we could avoid the hardcoding of the library names. What about this?
```
netcdf_libraries = spec['netcdf-c'].libs
if spec['hdf5'].satisfies('~shared'):
    netcdf_libraries += spec['hdf5'].libs
...
    ld_flags_from_library_list(netcdf_libraries)
```

If what I suggested doesn't work (modulo obvious typos) then I'm fine with what you have here.

Reply:
implemented a modified version of greg's..